### PR TITLE
Use new eslint dependency and remove tslint

### PR DESCRIPTION
**Pull request by JorikSchellekens**
This is expected to change with changes to https://github.com/matrix-org/eslint-config-matrix-org/pull/1 but this is a solid example of the new configuration.

Requires: https://github.com/matrix-org/pipelines/pull/77

### CODE REVIEW

**Comment by JorikSchellekens** [comment]
I don't know what needs to happen to CI for this change. Some guidance would be welcome

**Comment by turt2live** [comment]
Adjusting the appropriate pipelines in https://github.com/matrix-org/pipelines - if it passes locally then it's fine, though the CI team will generally merge quickly so ideally we only open a PR there when these switch PRs are approved.

File: src/stores/room-list/RoomListStore2.ts [hunk 1]
```diff
@@ -180,7 +180,9 @@ export class RoomListStore2 extends AsyncStore<ActionPayload> {
             const roomId = eventPayload.event.getRoomId();
             const room = this.matrixClient.getRoom(roomId);
             const tryUpdate = async (updatedRoom: Room) => {
-                console.log(`[RoomListDebug] Live timeline event ${eventPayload.event.getId()} in ${updatedRoom.roomId}`);
+                console.log(
+                    `[RoomListDebug] Live timeline event ${eventPayload.event.getId()} in ${updatedRoom.roomId}`,
+                );
```
Comment (turt2live) [inline src/stores/room-list/RoomListStore2.ts:185]: can we just bump the length limit or shorten the statement? 3 lines for debug line is a bit silly.

File: src/stores/room-list/algorithms/tag-sorting/RecentAlgorithm.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ import * as Unread from "../../../../Unread";
  * useful to the user.
  */
 export class RecentAlgorithm implements IAlgorithm {
-    public async sortRooms(rooms: Room[], tagId: TagID): Promise<Room[]> {
+    public async sortRooms(rooms: Room[], _: TagID): Promise<Room[]> {
```
Comment (turt2live) [inline src/stores/room-list/algorithms/tag-sorting/RecentAlgorithm.ts:28]: no thanks, let's keep the name even if unused.

Comment (JorikSchellekens) [reply src/stores/room-list/algorithms/tag-sorting/RecentAlgorithm.ts:28]: Ah, I fixed this in the rules but forgot to revert it. Thanks

File: src/DeviceListener.ts [hunk 3]
```diff
@@ -48,8 +48,8 @@ export default class DeviceListener {
     private displayingToastsForDeviceIds = new Set<string>();
 
     static sharedInstance() {
-        if (!window.mx_DeviceListener) window.mx_DeviceListener = new DeviceListener();
-        return window.mx_DeviceListener;
+        if (!window.mxDeviceListener) window.mxDeviceListener = new DeviceListener();
+        return window.mxDeviceListener;
```
Comment (JorikSchellekens) [inline src/DeviceListener.ts:52]: I hope these are alright to rename

File: src/indexing/BaseEventIndexManager.ts [hunk 4]
```diff
@@ -18,14 +18,14 @@ export interface MatrixEvent {
     type: string;
     sender: string;
     content: {};
-    event_id: string;
-    origin_server_ts: number;
+    eventId: string;
+    originServerTs: number;
```
Comment (JorikSchellekens) [inline src/indexing/BaseEventIndexManager.ts:22]: Not sure these are ok to rename either

File: src/indexing/BaseEventIndexManager.ts [hunk 5]
```diff
@@ -55,11 +55,11 @@ export interface SearchResult {
 }
 
 export interface SearchArgs {
-    search_term: string;
-    before_limit: number;
-    after_limit: number;
-    order_by_recency: boolean;
-    room_id?: string;
+    searchTerm: string;
```
Comment (JorikSchellekens) [inline src/indexing/BaseEventIndexManager.ts:58]: ditto

**Comment by JorikSchellekens** [comment]
This is silly but I can't find the button to undraft this

**Comment by JorikSchellekens** [comment]
Also, the conflicts are lies

File: src/editor/serialize.ts [hunk 6]
```diff
@@ -31,7 +31,8 @@ export function mdSerialize(model: EditorModel) {
                 return html + part.text;
             case "room-pill":
             case "user-pill":
-                return html + `[${part.text.replace(/[[\\\]]/g, c => "\\" + c)}](${makeGenericPermalink(part.resourceId)})`;
+                return html +
+                    `[${part.text.replace(/[[\\\]]/g, c => "\\" + c)}](${makeGenericPermalink(part.resourceId)})`;
```
Comment (turt2live) [inline src/editor/serialize.ts:35]: literally 1 character over :(

File: src/hooks/useAccountData.ts [hunk 7]
```diff
@@ -30,7 +30,7 @@ export const useAccountData = <T extends {}>(cli: MatrixClient, eventType: strin
     const handler = useCallback((event) => {
         if (event.getType() !== eventType) return;
         setValue(event.getContent());
-    }, [cli, eventType]);
+    }, [eventType]);
```
Comment (turt2live) [inline src/hooks/useAccountData.ts:33]: I don't know enough about hooks, but this and the one below feel wrong.

File: src/indexing/BaseEventIndexManager.ts (hunk 4 above)
Comment (turt2live) [reply src/indexing/BaseEventIndexManager.ts:22]: Very unsafe. They must remain underscored.

File: src/DeviceListener.ts (hunk 3 above)
Comment (turt2live) [reply src/DeviceListener.ts:52]: the window.* stuff should be fine

File: src/indexing/BaseEventIndexManager.ts (hunk 5 above)
Comment (turt2live) [reply src/indexing/BaseEventIndexManager.ts:58]: also unsafe

File: src/integrations/IntegrationManagers.ts [hunk 8]
```diff
@@ -123,7 +123,12 @@ export class IntegrationManagers {
             const apiUrl = data['api_url'];
             if (!apiUrl || !uiUrl) return;
 
-            const manager = new IntegrationManagerInstance(Kind.Account, apiUrl, uiUrl, w['id'] || w['state_key'] || '');
+            const manager = new IntegrationManagerInstance(
+                Kind.Account,
+                apiUrl,
+                uiUrl,
+                w['id'] || w['state_key'] || '',
+            );
```
Comment (turt2live) [inline src/integrations/IntegrationManagers.ts:131]: Also 1 character over 😭

File: src/stores/room-list/algorithms/Algorithm.ts [hunk 9]
```diff
@@ -382,7 +382,7 @@ export class Algorithm extends EventEmitter {
 
         if (!this._stickyRoom) {
             // If there's no sticky room, just do nothing useful.
-            if (!!this._cachedStickyRooms) {
+            if (this._cachedStickyRooms) {
```
Comment (turt2live) [inline src/stores/room-list/algorithms/Algorithm.ts:385]: Please can we keep the boolean coercion?

File: src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts [hunk 10]
```diff
@@ -257,6 +262,7 @@ export class ImportanceAlgorithm extends OrderingAlgorithm {
 
             if (indices[lastCat] > indices[thisCat]) {
                 // "should never happen" disclaimer goes here
+                // eslint-disable-next-line max-len
```
Comment (turt2live) [inline src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts:265]: why does this one get a max-len exclusion but the others don't?

File: src/hooks/useAccountData.ts (hunk 7 above)
Comment (JorikSchellekens) [reply src/hooks/useAccountData.ts:33]: The array specifies which values have to change before the hook is re-triggered. In both cases neither hook's value depends on the removed argument.

Comment (t3chguy) [reply src/hooks/useAccountData.ts:33]: Its fine, I think its from before I changed how it worked

File: src/stores/room-list/algorithms/Algorithm.ts (hunk 9 above)
Comment (JorikSchellekens) [reply src/stores/room-list/algorithms/Algorithm.ts:385]: Sure!

File: src/editor/serialize.ts (hunk 6 above)
Comment (JorikSchellekens) [reply src/editor/serialize.ts:35]: Yeah but that'll keep adding up and then my viming will be very clunky and sad

File: src/hooks/useAccountData.ts (hunk 7 above)
Comment (JorikSchellekens) [reply src/hooks/useAccountData.ts:33]: 👍

File: src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts (hunk 10 above)
Comment (JorikSchellekens) [reply src/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm.ts]: This one no longer does

File: src/indexing/BaseEventIndexManager.ts [hunk 11]
```diff
@@ -14,18 +14,21 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// The following interfaces take their names and member names from the js-sdk
```
Comment (turt2live) [inline src/indexing/BaseEventIndexManager.ts:17]:
```suggestion
// The following interfaces take their names and member names from seshat and the spec
```

**Pull request merged**